server/base/src/main/java/org/apache/accumulo/server/fs/VolumeManagerImpl.java

```diff
@@ -384,20 +390,34 @@ public short getDefaultReplication(Path path) {
    * @param filesystemURI Volume Filesystem URI
    * @return Hadoop Configuration with custom overrides for this FileSystem
    */
-  private static Configuration getVolumeManagerConfiguration(AccumuloConfiguration conf,
+  protected static Configuration getVolumeManagerConfiguration(AccumuloConfiguration conf,
       final Configuration hadoopConf, final String filesystemURI) {
 
-    final Configuration volumeConfig = new Configuration(hadoopConf);
+    Map<String,String> volumeHdfsConfigOverrides =
+        conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUME_CONFIG_PREFIX).entrySet()
+            .stream().filter(e -> e.getKey().startsWith(filesystemURI + ".")).collect(
+                Collectors.toUnmodifiableMap(e -> e.getKey().substring(filesystemURI.length() + 1),
+                    Entry::getValue));
 
-    conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUME_CONFIG_PREFIX).entrySet()
-        .stream().filter(e -> e.getKey().startsWith(filesystemURI + ".")).forEach(e -> {
-          String key = e.getKey().substring(filesystemURI.length() + 1);
-          String value = e.getValue();
-          log.info("Overriding property {} for volume {}", key, value, filesystemURI);
-          volumeConfig.set(key, value);
+    if (volumeHdfsConfigOverrides.isEmpty()) {
+      // There are no overrides in the AccumuloConfiguration for this volume, return the Hadoop
+      // Configuration input parameter.
+      return hadoopConf;
+    } else {
+      // Get the Hadoop Configuration object from the Cache for this filesystemURI. If no object
+      // exists, then create one, apply the overrides and put it in the cache. The Volume config
+      // overrides cannot change, so we only need to cache this once
+      return HDFS_CONFIGS_FOR_VOLUME.get(filesystemURI, (fs) -> {
```

Comment:
This is a bit hacky, but it seems like it could work to support different hadoopConfig instances being passed in.  When the exact same object is passed in should use the cached entry.

```suggestion
    /**
     * Hadoop does not implement hashCode for the configuration class, so the following gets the
     *  objects hashcode.  The concept is to make the cached object have 1:1 correspondence with
     *  the hadoop config object from which it was derived.
     */
     String key = filesystemURI+hadoopConf.hashCode();

      return HDFS_CONFIGS_FOR_VOLUME.get(key, (fs) -> {
```

Reply:
I think there is only one Hadoop configuration object in use. I didn't track this down though and instead just implemented your suggestion in 1b2fb52.